cmd/skaffold/app/cmd/diagnose.go

```diff
@@ -46,22 +52,13 @@ func NewCmdDiagnose() *cobra.Command {
 }
 
 func doDiagnose(ctx context.Context, out io.Writer) error {
-	runCtx, configs, err := runContext(out, opts)
+	configs, err := withFallbackConfigFunc(out, opts, parser.GetAllConfigs)
```

Comment:
I think that `withFallbackConfigFunc` is trigger the flow of auto-generating `skaffold.yaml` if it doesn't exist (auto skaffold init). We could call `parser.GetAllConfigs` directly for `skaffold diagnose`?